# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1397695440 14400
#      Wed Apr 16 20:44:00 2014 -0400
# Node ID 3647db1a37d7513b32f37a3723412f0721f3e52b
# Parent  491b0adfec957983032846a20d993160362c57db
NEWS entry for previous printf format changes

diff --git a/NEWS b/NEWS
--- a/NEWS
+++ b/NEWS
@@ -4,16 +4,31 @@ Summary of important user-visible change
  ** Interpolation function changes for Matlab compatibility
 
     The interpolation method 'cubic' is now equivalent to 'pchip'
     for interp1, interp2, and interp3.  Previously, 'cubic' was equivalent
     to 'spline' for interp2.  This may produce different results as 'spline'
     has continuous 1st and 2nd derivatives while 'pchip' only has a continuous
     1st derivative.
 
+ ** Integer formats used in the printf family of functions now work for
+    64-bit integers and are more compatible with Matlab when printing
+    non-integer values.  Now instead of truncating, Octave will switch
+    the effective format to '%g' in the following circumstances:
+
+      * the value of an integer type (int8, uint32, etc.) value exceeds
+        the maximum for the format specifier.  For '%d', the limit is
+        intmax ('int64') and for '%u' it is intmax ('uint64').
+
+      * round(x) != x or the value is outside the range allowed by the
+        integer format specifier.
+
+    There is still one difference:  Matlab switches to '%e' and Octave
+    is currently switching to '%g'.
+
  ** Other new functions added in 4.2:
 
       dir_in_loadpath
       hgload
       hgsave
       numfields
 
  ** Deprecated functions.
